src/js/models/Artefacts.js

```diff
@@ -0,0 +1,60 @@
+import MapList from './MapList.js'
+import Artefact from './Artefact.js'
+import axios from 'axios'
+import {wktPolygonToSvg, dbMatrixToSVG} from '~/utils/VectorFactory.js'
+
+export default class Artefacts extends MapList {
+
+  constructor(
+    sessionID,
+    idKey,
+    ajaxPayload = undefined,
+    attributes = {})
+  {
+    idKey = idKey || 'artefact_position_id'
+    ajaxPayload = ajaxPayload ? ajaxPayload : {transaction: 'getArtOfImage'}
+    super(sessionID, idKey, ajaxPayload, Artefact, attributes)
+  }
+
+  static getModel() {
+    return Artefact
+  }
+
+  // We should eventually have a hash associated with each mask,
+  // and should be prepared to recieve a message back from the server
+  // saying "nothing changed" and then we can leave the artefact alone.
+  fetchMask(scrollVersionID, artefactID) {
+    scrollVersionID = scrollVersionID >>> 0
+    artefactID = artefactID >>> 0
+    let payload = {
+      transaction: 'getArtefactMask',
+      SESSION_ID: this.sessionID,
+      scrollVersion: scrollVersionID,
+      artID: artefactID
+    }
+
+    return new Promise((resolve, reject) => {
+      try {
```

Comment:
the `try/catch` should probably be inside the `then` block and a `.catch(reject)` in there as well if the request fails.